File: account-service/src/main/java/com/food/delivery/accountservice/company/CompanyGetService.java
package com.food.delivery.accountservice.company;

import com.food.delivery.accountservice.employee.EmployeeRepository;
import com.food.delivery.accountservice.employee.domain.Employee;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;
import reactor.core.publisher.Mono;

@Service
@AllArgsConstructor
public class CompanyGetService {

	private final CompanyClient companyClient;
	private final EmployeeRepository employeeRepository;

	public Mono<CompanyRest> getCompany(String employeeEmail) {
		return employeeRepository.findByEmail(employeeEmail)
				.map(Employee::getCompanyId)
				.flatMap(companyClient::getCompany);
	}
}

File: menu-service/src/main/kotlin/com/food/delivery/menuservice/company/CompanyGetService.kt
package com.food.delivery.menuservice.company

import com.food.delivery.menuservice.employee.EmployeeClient
import org.springframework.stereotype.Service
import reactor.core.publisher.Mono

@Service
class CompanyGetService(private val employeeClient: EmployeeClient, private val companyClient: CompanyClient) {

    fun getCompany(employeeAdminEmail: String): Mono<CompanyRest> {
        return employeeClient.getEmployee(employeeAdminEmail)
                .flatMap { companyClient.getCompany(it.companyId) }
    }
}
